api/src/services/utils.js

```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  parseQueryParams: query => {
+    const parsed = {};
+    if (!query || typeof query !== 'object') {
+      return parsed;
+    }
+
+    Object.keys(query).forEach(key => {
+      try {
+        parsed[key] = JSON.parse(query[key]);
+      } catch(e) {
+        parsed[key] = query[key];
+      }
+    });
+
+    return parsed;
+  }
+};
```

Comment:
I think this is being called too low in the stack - the services shouldn't have to worry about the formatting of the request parameters. This could be moved up to the controllers but I think it's equivalent to a body parser so maybe it should be a middleware, so we can turn it on on specific endpoints and it'll just work. What do you think?

Reply:
Yea, fully agree! Middleware created!